toolparameteriser/testcreator.py

```diff
@@ -56,19 +76,33 @@ def _get_tmpl_values(self,parameters:dict,work_dir:str)->dict:
         return params
 
     def _run_job(self,parameters:dict,runID:str,work_dir:str):
+
         #Prepare values for tmpl
+        logging.debug("Preparing parameters for job template.")
         params=self._get_tmpl_values(parameters,work_dir) 
+        logging.debug("Successfully prepared job parameters.")
+
         #Substitute Tmpl 
+        logging.debug("Inserting job parameters into job template.")
         with open(os.path.join(self.Config["Output_path"],self.tmplfile), 'r') as f:
             template = Template(f.read())
             result = template.safe_substitute(params)
+        logging.debug("Successfully inserted job parameters into job template.")
+
         #Save to file
-        with open(os.path.join(self.Config["Output_path"],runID,"batch.slurm"), 'w') as f:
+        scriptdir = os.path.join(self.Config["Output_path"],runID)
+        scriptpath = os.path.join(self.Config["Output_path"],runID,"batch.slurm")
+        with open(scriptpath, 'w') as f:
             f.write(result)
+        logging.info(f"Saved job script to {scriptpath}.")
+
         #RUN if not dryrun
-        if not self.Config["dryrun"]:
-                os.system(f"cd {os.path.join(self.Config['Output_path'],runID)};"+ 
-                    f"sbatch {os.path.join(self.Config['Output_path'],runID,'batch.slurm')}")
+        if self.Config["dryrun"]:
+            logging.info(f"sbatch --chdir={scriptdir} {scriptpath}")
+        else:
+            # os.system(f"cd {scriptdir} && " + f"sbatch {scriptpath}")
+            msg = subprocess.check_output(["sbatch", f"--chdir={scriptdir}", scriptpath], stderr=subprocess.STDOUT)
```

Comment:
`subprocess.run` is recommended instead of `.check_output`

Reply:
Could you elaborate on why run is preferred? I believe check_output is equivalent to `run(..., check=True, stdout=PIPE).stdout` https://docs.python.org/3/library/subprocess.html#subprocess.check_output

Reply:
Yep, basically. It's just a more flexible/powerful function. I'm just basing this on what the docs say

> The recommended approach to invoking subprocesses is to use the [run()](https://docs.python.org/3/library/subprocess.html#subprocess.run) function for all use cases it can handle. [[1](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module)]

Reply:
resolved in https://github.com/WEHI-ResearchComputing/ToolParameteriser/pull/15/commits/27644cef72853667508e259d45dc6c4232e40f79